python/taichi/ui/gui.py

```diff
@@ -13,6 +14,30 @@ class Gui:
     def __init__(self, gui) -> None:
         self.gui = gui  #reference to a PyGui
 
+    @contextmanager
+    def sub_window(self, name, x, y, width, height):
+        """Creating a context manager for subwindow
+
+        Note::
+            All args of this method should align with `begin`.
+
+        Args::
+            x (float): The x-coordinate (between 0 and 1) of the top-left corner of the subwindow, relative to the full window.
+            y (float): The y-coordinate (between 0 and 1) of the top-left corner of the subwindow, relative to the full window.
+            width (float): The width of the subwindow relative to the full window.
+            height (float): The height of the subwindow relative to the full window.
+
+        Usage::
```

Comment:
About the single `:` versus double `::` situation, we only need to use the double `::` before a code block. So ti would still be `Args:`, but `Usage::` :-)

Reply:
Got.